src/components/ImageUpload/index.jsx

```diff
@@ -10,22 +10,39 @@ class ImageUpload extends React.Component {
     super(props);
     this.state = {
       value: props.input.value,
+      sizeValidationError: false,
     };
 
     this.handleFilePicked = this.handleFilePicked.bind(this);
-  }
-
-  updateValue(value) {
-    this.setState({ value });
-    this.props.input.onChange(value);
+    this.sizeValidator = this.sizeValidator.bind(this);
   }
 
   handleFilePicked(event) {
     const reader = new FileReader();
+    const file = event.target.files[0];
+    // File size is given in bytes
+    const oneMegabyte = 1048576;
+
+    if (file && file.size > oneMegabyte) {
+      this.setState({ sizeValidationError: true });
+      return;
+    }
+
     reader.onload = () => {
-      this.updateValue(reader.result);
+      this.setState({ value: reader.result });
     };
-    reader.readAsDataURL(event.target.files[0]);
+    reader.readAsDataURL(file);
+  }
+
+  sizeValidator(event) {
+    const { onChange } = this.props.input;
+    if (event.target.naturalHeight !== 675 || event.target.naturalWidth !== 1134) {
+      this.setState({ value: '', sizeValidationError: true });
+      onChange('');
```

Comment:
These two lines are to null out the value of the invalid input so we stop showing the image on the edit form and we won't send it with the form data. The input element will still show the file name of the failed upload.